ood-portal-generator/templates/ood-portal.conf.erb

```diff
@@ -63,12 +63,17 @@ Listen <%= addr_port %>
 # The Open OnDemand portal VirtualHost
 #
 <VirtualHost *:<%= @port %>>
+  <%- if @logformat -%>
+  LogFormat <%= @logformat %>
+  TransferLog "<%= @accesslog %>"
+  <% else %>
+  CustomLog "<%= @accesslog %>" combined
+  <%- end -%>
+  ErrorLog  "<%= @errorlog %>"
+
   <%- if @servername -%>
   ServerName <%= @servername %>
 
-  ErrorLog  "<%= @logroot %>/<%= @servername %>_error<%= "_ssl" if @ssl %>.log"
```

Comment:
I'd recommend trying to keep the location of the CustomLog and ErrorLog in same place by default because if you change location that will trigger Puppet at sites using our Puppet module to have to restart Apache which we try to avoid doing unnecessarily.

Reply:
The problem here is that if the `servername` is undefined then the Error and Custom log directives won't be configured properly. Additionally the whole logic of including `ssl` and `servername` strings into the log filename has been moved to `view.rb`.

Reply:
So the way around that is this pseudo code:

```
<%- if @servername -%>
ServerName ...
<%- end -%>

ErrorLog
CustomLog
<new log logic, etc>

<%- if @servername && @use_rewrites -%>
RewriteEngine On
...rewrite rules
<%- end -%>
```

This is also likely why the tests are failing because we test the rendered apache config and if the location of directives is changed then the tests will fail, the above should fix the tests if spacing is correct and also ensure that sites don't get Apache changes from this without them actually changing their YAML.